renew/mail.go

```diff
@@ -0,0 +1,150 @@
+package renew
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strconv"
+	"time"
+
+	"cloud.google.com/go/bigquery"
+	"github.com/wopta/goworkspace/lib"
+	"github.com/wopta/goworkspace/mail"
+	"github.com/wopta/goworkspace/models"
+)
+
+type RenewReq struct {
+	Date            string `json:"date"`
+	DaysBeforeRenew string `json:"days_before_renew"`
+}
+
+type MailReport struct {
+	Policy       string                `bigquery:"policyUid"`
+	Name         string                `bigquery:"name"`
+	Email        string                `bigquery:"email"`
+	CreationDate bigquery.NullDateTime `bigquery:"creationDate"`
+	MailError    string                `bigquery:"mailError"`
+}
+
+func RenewMailFx(w http.ResponseWriter, r *http.Request) (string, interface{}, error) {
+	var (
+		err             error
+		date            = time.Now().UTC()
+		daysBeforeRenew = 10
+		req             RenewReq
+	)
+
+	log.SetPrefix("[RenewMailFx] ")
+	defer func() {
+		collectionPrefix = ""
+		if err != nil {
+			log.Printf("error: %s", err.Error())
+		}
+		log.Println("Handler end -------------------------------------------------")
+	}()
+	log.Println("Handler start -----------------------------------------------")
+
+	body := lib.ErrorByte(io.ReadAll(r.Body))
+	defer r.Body.Close()
+	err = json.Unmarshal(body, &req)
+	if err != nil {
+		log.Printf("error unmarshaling request: %s", err.Error())
+		return "", nil, err
+	}
+
+	if req.Date != "" {
+		tmpDate, err := time.Parse(time.DateOnly, req.Date)
+		if err != nil {
+			log.Printf("error parsing request date: %s", err.Error())
+			return "", nil, err
+		}
+		date = tmpDate
+	}
+
+	if req.DaysBeforeRenew != "" {
+		tmpDays, err := strconv.Atoi(req.DaysBeforeRenew)
+		if err != nil {
+			log.Printf("error parsing target date: %s", err.Error())
+			return "", nil, err
+		}
+		daysBeforeRenew = tmpDays
+	}
+
+	targetDate := date.AddDate(0, 0, daysBeforeRenew)
+
+	policies, err := getRenewPolicies(targetDate)
+	if err != nil {
+		log.Printf("error getting renew policies: %s", err.Error())
+		return "", nil, err
+	}
+
+	for _, policy := range policies {
+		from := mail.AddressAnna
+		to := mail.GetContractorEmail(&policy)
+		flowName := models.ECommerceFlow
+		log.Printf("Sending email from %s to %s", from, to)
+		//err = writeMailError(policy.Uid, to.Name, to.Address, lib.GetBigQueryNullDateTime(time.Now().UTC()), "test error")
+		err = mail.SendMailRenewDraftV2(policy, from, to, mail.Address{}, flowName, policy.HasMandate)
```

Comment:
as commented on the mail part, remove the last argument since we pass the whole policy into the function